### ISSUE TITLE
Clarify relation to eslint, .eslintignore

### ISSUE THREAD

**Issue by joelpurra** [mozilla/addons-linter#3389]
### Describe the problem and steps to reproduce it:

- Create an extension which has a javascript build step using, for example, Typescript or Rollup.
- Build the extension, outputting files in the, for example, `dist/` directory.
- Add [`eslint`](https://eslint.org/) for source code linting.
- Create an [`.eslintignore` file](https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories) ignoring build artifacts.
- Run `npx addons-linter .`.

### What happened?

(Tableized CLI output.)

| Code | Message | Description | File | Line | Column |
| --- | --- | --- | --- | --- | --- |
File ignored because of a matching ignore pattern. Use "--no-ignore" to override. | File ignored because of a matching ignore pattern. Use "--no-ignore" to override. | dist/my-build-output.js |

### What did you expect to happen?

Built javascript files are linted for potential addon-related problems. (No code style-related linting.)

### Anything else we should know?

```shell
npx addons-linter --version
```

```text
2.7.0
```

Also tested using an installed `addons-linter` (as opposed to using `npx`) and with `web-ext` v5.2.0.

- The relation to `--no-ignore`, which seems to come from the `eslint` CLI, is unclear. The flag does not seem to work with `addons-linter` nor `web-ext`.
- The `npm`-packaged versions of `addons-linter`/`web-ext` do not contain a `.eslintignore` file, so it should not be interfering.
- Presumably unrelated to the files (un)ignored by default by `addons-linter`/`web-ext`, since `dist/` is not one of them.
  - https://github.com/mozilla/addons-linter/blob/2.8.0/src/scanners/javascript.js#L97
  - https://github.com/mozilla/web-ext/blob/5.2.0/src/util/file-filter.js#L40-L47
- Using `--ignore-files` to ignore files when using `addons-linter`/`web-ext` seems more appropriate.
- Using `web-ext --source-dir` from a directory with `.eslintignore` seems to apply the ignores also to the intended source directory.

**Comment by rpl** [comment]
The .eslintignore file should be actually ignored by addons-linter.

@willdurand This is very likely a regression introduced in v2.0.0 (likely by migrating from eslint 5 to eslint 7), as an additional proof `npx addons-linter@1.26.0 .` doesn't produce those warnings (while any version >= 2.0.0 does).

we should:
- look into the changes needed to make sure that we are configuring eslint v7 to ignore the .eslintignore
- add a new test case to cover the fix and prevent it from regressing again (because clearly there wasn't coverage for this scenario)
